File: c++/20366.cpp
#include <algorithm>
#include <iostream>
#include <vector>

using namespace std;

int main() {
  freopen("example.txt", "r", stdin);

  int N;
  cin >> N;

  vector<int> v;
  int minV = 1e9;

  for (int i = 0; i < N; i++) {
    int a;
    cin >> a;
    v.push_back(a);
  }

  sort(v.begin(), v.end());

  for (int i = 0; i < N; i++) {
    for (int j = i + 1; j < N; j++) {
      int snowman1 = v[i] + v[j];

      int start = 0;
      int end = N - 1;

      while (start < end) {
        if (start == i || start == j) {
          start++;
          continue;
        }
        if (end == i || end == j) {
          end--;
          continue;
        }

        int snowman2 = v[start] + v[end];
        minV = min(minV, abs(snowman1 - snowman2));

        if (snowman1 > snowman2)
          start++;
        else if (snowman1 < snowman2)
          end--;
        else {
          cout << 0;
          return 0;
        }
      }
    }
  }

  cout << minV;

  return 0;
}

File: js/20366.js
const fs = require('fs');

const inputs = fs.readFileSync('example.txt').toString().trim().split('\n');

const N = +inputs[0];
const arr = inputs[1]
  .split(' ')
  .map(Number)
  .sort((a, b) => a - b);

let min = Infinity;

search: for (let i = 0; i < N; i++) {
  for (let j = i + 1; j < N; j++) {
    const snowMan1 = arr[i] + arr[j];
    let start = 0;
    let end = N - 1;
    while (start < end) {
      if (start === i || start === j) {
        start++;
        continue;
      }
      if (end === i || end === j) {
        end--;
        continue;
      }

      const snowMan2 = arr[start] + arr[end];
      min = Math.min(min, Math.abs(snowMan1 - snowMan2));

      if (snowMan1 > snowMan2) start++;
      else if (snowMan1 < snowMan2) end--;
      else {
        min = 0;
        break search;
      }
    }
  }
}

console.log(min);
